src/ip/QosConfig.cc

```diff
@@ -325,13 +326,11 @@ Ip::Qos::Config::parseConfigLine()
                        << "Netfilter mark preservation not available.");
 #endif // USE_LIBNETFILTERCONNTRACK
 #elif SO_MARK // SO_MARK && USE_LIBCAP
-                debugs(3, DBG_CRITICAL, "ERROR: Invalid parameter 'mark' in qos_flows option. "
-                       << "Linux Netfilter marking not available without LIBCAP support.");
-                self_destruct();
+                throw TextException(ToSBuf("Invalid parameter 'mark' in qos_flows option. ",
+                                           "Linux Netfilter marking not available on this platform."), Here());
 #else // SO_MARK && USE_LIBCAP
-                debugs(3, DBG_CRITICAL, "ERROR: Invalid parameter 'mark' in qos_flows option. "
-                       << "Linux Netfilter marking not available on this platform.");
-                self_destruct();
+                throw TextException(ToSBuf("Invalid parameter 'mark' in qos_flows option. ",
+                                           "Linux Netfilter marking not available on this platform."), Here());
```

Comment:
We do not really need ToSBuf() in this and similar contexts where just string literals are concatenated[^1], but since other nearby exceptions have to use ToSBuf(), I think it is OK to use ToSBuf() here as well. Your call.

[^1]: Removing a comma between the two arguments will allow C++ to concatenate [string literals](https://en.cppreference.com/w/cpp/language/string_literal) instead of ToSBuf() doing the same at runtime AFAICT.